src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -360,25 +274,120 @@ void HierRTLMP::hierRTLMacroPlacer()
     if (min_num_macro_base_ <= 0)
       min_num_macro_base_ = 1;  // lowerbound
     max_num_macro_base_ = min_num_macro_base_ * coarsening_ratio_ / 2.0;
-
     if (metrics_->getNumMacro() <= 150) {
-      max_num_level_
-          = 1;  // if the number of macros is small we can do single level
+      // If the number of macros is less than the threshold value, reset to
+      // single level.
+      max_num_level_ = 1;
+      logger_->report("Reset number of levels to 1");
     }
   }
 
+  //
+  // Set sizes for root level based on coarsening_factor and the number of
+  // physical hierarchy levels
+  //
   unsigned coarsening_factor = std::pow(coarsening_ratio_, max_num_level_ - 1);
   max_num_macro_base_ = max_num_macro_base_ * coarsening_factor;
   min_num_macro_base_ = min_num_macro_base_ * coarsening_factor;
   max_num_inst_base_ = max_num_inst_base_ * coarsening_factor;
   min_num_inst_base_ = min_num_inst_base_ * coarsening_factor;
+
   logger_->report(
       "num level: {}, max_macro: {}, min_macro: {}, max_inst:{}, min_inst:{}",
       max_num_level_,
       max_num_macro_base_,
       min_num_macro_base_,
       max_num_inst_base_,
       min_num_inst_base_);
+}
+
+///////////////////////////////////////////////////////////////
+// Top Level Interface function
+//
+void HierRTLMP::hierRTLMacroPlacer()
+{
+  //
+  // Get the database information
+  //
+  block_ = db_->getChip()->getBlock();
+  dbu_ = db_->getTech()->getDbUnitsPerMicron();
+  // report the parameters
+  logger_->report("area_weight_ = {}", area_weight_);
+  logger_->report("outline_weight_ = {}", outline_weight_);
+  logger_->report("wirelength_weight_ = {}", wirelength_weight_);
+  logger_->report("guidance_weight_ = {}", guidance_weight_);
+  logger_->report("fence_weight_ = {}", fence_weight_);
+  logger_->report("boundary_weight_ = {}", boundary_weight_);
+  logger_->report("notch_weight_ = {}", notch_weight_);
+  logger_->report("macro_blockage_weight_ = {}", macro_blockage_weight_);
+  logger_->report("halo_width_ = {}", halo_width_);
+
+  // calculate the pitch_x_ and pitch_y automatically based on macro pins
+  // pitch_x_ = dbuToMicron(
```

Comment:
It looks like this was just moved but is there a plan to turn it on?

Reply:
already turned on -- deleted the commented code